lib/mayaUsd/ufe/UsdTransform3dMayaXformStack.cpp

```diff
@@ -0,0 +1,708 @@
+// UsdMayaXformStack::IsCompatibleType().  Just roll our own op name to
+// Maya transform stack index position.
+const std::unordered_map<std::string, UsdTransform3dMayaXformStack::OpNdx> opNameToNdx{
+    {"xformOp:translate",           UsdTransform3dMayaXformStack::NdxTranslate},
+    {"xformOp:translate:rotatePivotTranslate", UsdTransform3dMayaXformStack::NdxRotatePivotTranslate},
+    {"xformOp:translate:rotatePivot", UsdTransform3dMayaXformStack::NdxRotatePivot},
+    {"xformOp:rotateX",             UsdTransform3dMayaXformStack::NdxRotate},
+    {"xformOp:rotateY",             UsdTransform3dMayaXformStack::NdxRotate},
+    {"xformOp:rotateZ",             UsdTransform3dMayaXformStack::NdxRotate},
+    {"xformOp:rotateXYZ",           UsdTransform3dMayaXformStack::NdxRotate},
+    {"xformOp:rotateXZY",           UsdTransform3dMayaXformStack::NdxRotate},
+    {"xformOp:rotateYXZ",           UsdTransform3dMayaXformStack::NdxRotate},
+    {"xformOp:rotateYZX",           UsdTransform3dMayaXformStack::NdxRotate},
+    {"xformOp:rotateZXY",           UsdTransform3dMayaXformStack::NdxRotate},
+    {"xformOp:rotateZYX",           UsdTransform3dMayaXformStack::NdxRotate},
+    {"xformOp:orient",              UsdTransform3dMayaXformStack::NdxRotate},
+    {"xformOp:rotateXYZ:rotateAxis", UsdTransform3dMayaXformStack::NdxRotateAxis},
+    {"!invert!xformOp:translate:rotatePivot", UsdTransform3dMayaXformStack::NdxRotatePivotInverse},
+    {"xformOp:translate:scalePivotTranslate", UsdTransform3dMayaXformStack::NdxScalePivotTranslate},
+    {"xformOp:translate:scalePivot", UsdTransform3dMayaXformStack::NdxScalePivot},
+    {"xformOp:shear",               UsdTransform3dMayaXformStack::NdxShear},
+    {"xformOp:scale",               UsdTransform3dMayaXformStack::NdxScale},
+    {"!invert!xformOp:translate:scalePivot", UsdTransform3dMayaXformStack::NdxScalePivotInverse}};
+
+//----------------------------------------------------------------------
+// Conversion functions from RotXYZ to all supported rotation attributes.
+//----------------------------------------------------------------------
+
+typedef VtValue (*CvtRotXYZToAttrFn)(double x, double y, double z);
+
+VtValue toXYZ(double x, double y, double z) {
+    // No rotation order conversion
+    VtValue v;
+    v = GfVec3f(x, y, z);
+    return v;
+}
+
+// Reorder argument RotXYZ rotation.
+template<MEulerRotation::RotationOrder DST_ROT_ORDER>
+VtValue to(double x, double y, double z) {
+    MEulerRotation eulerRot(TO_RAD(x), TO_RAD(y), TO_RAD(z), MEulerRotation::kXYZ);
+    eulerRot.reorderIt(DST_ROT_ORDER);
+    VtValue v;
+    v = GfVec3f(TO_DEG(eulerRot.x), TO_DEG(eulerRot.y), TO_DEG(eulerRot.z));
+    return v;
+}
+
+auto toXZY = to<MEulerRotation::kXZY>;
+auto toYXZ = to<MEulerRotation::kYXZ>;
+auto toYZX = to<MEulerRotation::kYZX>;
+auto toZXY = to<MEulerRotation::kZXY>;
+auto toZYX = to<MEulerRotation::kZYX>;
+
+VtValue toX(double x, double, double) {
```

Comment:
These RotXYZ to single-axis rotation conversion functions don't work, much to my chagrin.  Have not debugged them seriously, but when used the object actually moves along two axes!